WiseUnpacker/SabreTools.Serialization.Deserializers/WiseSectionHeader.cs

```diff
@@ -87,39 +87,193 @@ public class WiseSectionHeader : BaseBinaryDeserializer<SectionHeader>
+            header.UnknownDataSize = data.ReadUInt32LittleEndian();
             header.SecondExecutableFileEntryLength = data.ReadUInt32LittleEndian();
             header.UnknownValue2 = data.ReadUInt32LittleEndian();
             header.UnknownValue3 = data.ReadUInt32LittleEndian();
             header.UnknownValue4 = data.ReadUInt32LittleEndian();
             header.FirstExecutableFileEntryLength = data.ReadUInt32LittleEndian();
             header.MsiFileEntryLength = data.ReadUInt32LittleEndian();
-            if (headerLength == 6)
-                return header;
+            if (headerLength != 6)
+            {
+                header.UnknownValue7 = data.ReadUInt32LittleEndian();
+                header.UnknownValue8 = data.ReadUInt32LittleEndian();
+                if (headerLength != 8)
+                {
+                    header.ThirdExecutableFileEntryLength = data.ReadUInt32LittleEndian();
+                    header.UnknownValue10 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue11 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue12 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue13 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue14 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue15 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue16 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue17 = data.ReadUInt32LittleEndian();
+                    if (headerLength != 17)
+                    {
+                        header.UnknownValue18 = data.ReadUInt32LittleEndian();
+                    }
+                }
+            }
+            
 
-            header.UnknownValue7 = data.ReadUInt32LittleEndian();
-            header.UnknownValue8 = data.ReadUInt32LittleEndian();
-            if (headerLength == 8)
-                return header;
+            // Parse strings
+            // TODO: Count size of string section for later size verification
+            
+            data.Seek(initialOffset + localWisOffset, 0);
+            header.TmpString = data.ReadNullTerminatedString(Encoding.ASCII); // read .TMP string
+            header.GuidString = data.ReadNullTerminatedString(Encoding.ASCII); // read GUID string
+            // TODO: better way to figure out how far it's needed to advance?
+            if (header.Version == null)
+            {
+                // TODO: error output
+                return null;
+            }
+            int versionSize;
+            if (header.Version[header.Version.Length - 1] == 0x02)
+            {
+                versionSize = header.Version[header.Version.Length - 3];
+            }
+            else
+            {
+                versionSize = header.Version[header.Version.Length - 2];
+            }
+            if (versionSize != 1) // third byte seems to indicate size of NonWiseVer
```

Comment:
As with a comment above, I strongly encourage "range" checks, not just `!=`. If you mean that the value should be `> 1`, please use that instead.

Reply:
In this situation, I specifically want to make sure that the version size is not equal to 1, every other number would be valid. Besides 0 I guess, but no samples I have, not even the earliest ones, have this set to 0. It wouldn't harm functionality to check if the value is 1 or lower, I guess, so if you want me to do that, I can?